CMakeLists.txt

```diff
@@ -0,0 +1,61 @@
+#######################################
+# General
+cmake_minimum_required(VERSION 3.10)
+
+project(libmicrofido2 C)
+set(PRODUCT_NAME microfido2)
+
+#######################################
+# Compilation
+
+add_compile_options(
+    -std=gnu99
+    -fstack-usage
+    -ffunction-sections
+    -fdata-sections
+    -fshort-enums
+    -Wall
+    -Werror
+)
+
+IF(CMAKE_BUILD_TYPE MATCHES Debug)
+    add_compile_options(
+        -ggdb
+        -O0
+    )
+ELSE()
+    add_compile_options(
+        -Os
+    )
+ENDIF(CMAKE_BUILD_TYPE MATCHES Debug)
```

Comment:
Is it better style to include the if-statement in the endif clause? I saw others just use `endif()`. Also the [CMake docs](https://cmake.org/cmake/help/latest/command/endif.html) states

> The optional <condition> argument is supported for backward compatibility only. If used it must be a verbatim repeat of the argument of the opening if clause.